core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -362,10 +363,7 @@ public final RunT getPreviousBuild() {
 
                 if (r == null) {
                     // having two neighbors pointing to each other is important to make RunMap.removeValue work
-                    JobT _parent = asRun().getParent();
-                    if (_parent == null) {
-                        throw new IllegalStateException("no parent for " + asRun().number);
-                    }
+                    JobT _parent = Objects.requireNonNull(asRun().getParent(), "no parent for " + asRun().number);
```

Comment:
A more direct way of expressing the same concept that also aids the static analyzer.